### PR TITLE
WIP schema codec using apache avro schema impl

### PR DESCRIPTION
**Pull request by TobiasPfeifer**
solves #188 
enables to continue work on #144 

This implementation builds on top of the Apache Avro implementation and uses their Schema as an intermediate representation.

### CODE REVIEW

File: zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala [hunk 1]
```diff
@@ -0,0 +1,2020 @@
...
+        assert(schema)(isRight(isStandardType(StandardType.UnitType)))
+      }
+    ),
+    test("encode/decode full adt test") {
+      val initialSchemaDerived = DeriveSchema.gen[FullAdtTest.TopLevelUnion]
+
+      val decoded = for {
+        avroSchemaString <- AvroSchemaCodec.encode(initialSchemaDerived)
+        decoded          <- AvroSchemaCodec.decode(Chunk.fromArray(avroSchemaString.getBytes()))
+        //_ <- AvroSchemaCodec.encode(decoded) TODO: this fails
```
Comment (TobiasPfeifer) [inline zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala:1728]: this needs to be fixed

File: zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala [hunk 2]
```diff
@@ -0,0 +1,2020 @@
...
+        unit: Unit,
+        char: Char,
+        uuid: UUID
+      ) extends TopLevelUnion
+      case class NestedRecord(innerRecord: InnerRecord)               extends TopLevelUnion
+      case class Unions(union: Union)                                 extends TopLevelUnion
+      case class Enumeration(enum: Enum)                              extends TopLevelUnion
+      case class Iterables(list: List[String], map: Map[String, Int]) extends TopLevelUnion
+
+      // TODO: Schema derivation fails for the following case classes
```
Comment (TobiasPfeifer) [inline zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala:1998]: is there an open issue for this already (bug or feature)?

Comment (thinkharderdev) [reply zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala:1998]: The java time schemas take a formatter so we only provide default implicit for these in the `zio.schema.DefaultJavaTimeSchemas` trait. So you need to mix that trait in for the implicit derivation to work here I believe.

**Comment by jdegoes** [comment]
@TobiasPfeifer Any chance you will get a chance to finish this? 🙏 

Will close for now, please re-open if so!

**Pull request closed without merging**